macros/shared/src/test/scala-2.x/zio/macros/AccessibleSpec.scala

```diff
@@ -117,6 +117,79 @@ object AccessibleSpec extends DefaultRunnableSpec {
           """
         })(isRight(anything))
       },
+      testM("generates accessor for service with one type param") {
+        assertM(typeCheck {
+          """
+             @accessible
+             object Module {
+               trait Service[T] {
+                 val v: Task[T]
+                 def f1: UIO[Unit]
+                 def f2(): UIO[Unit]
+                 def f3(t: T): UIO[Unit]
+                 def f4(t: T)(i: Int): UIO[Unit]
+                 def f5(t: T)(implicit i: Int): UIO[Unit]
+                 def f6(t: T*): UIO[Unit]
+               }
+             }
+
+             object Check {
+               def v[T: Tag]: ZIO[Has[Module.Service[T]], Throwable, T] =
+                 Module.v[T]
+               def f1[T: Tag]: ZIO[Has[Module.Service[T]], Nothing, Unit] =
+                 Module.f1[T]
+               def f2[T: Tag](): ZIO[Has[Module.Service[T]], Nothing, Unit] =
+                 Module.f2[T]()
+               def f3[T: Tag](t: T): ZIO[Has[Module.Service[T]], Nothing, Unit] =
+                 Module.f3[T](t)
+               def f4[T: Tag](t: T)(i: Int): ZIO[Has[Module.Service[T]], Nothing, Unit] =
+                 Module.f4[T](t)(i)
+               def f5[T: Tag](t: T)(implicit i: Int): ZIO[Has[Module.Service[T]], Nothing, Unit] =
+                 Module.f5[T](t)
+               def f6[T: Tag](t: T*): ZIO[Has[Module.Service[T]], Nothing, Unit] =
+                 Module.f6[T](t: _*)
+             }
+          """
+        })(isRight(anything))
+      },
+      testM("generates accessor for service with two type params and type bounds") {
+        assertM(typeCheck {
+          """
+             trait Foo
+             trait Bar
+
+             @accessible
+             object Module {
+               trait Service[T <: Foo, U >: Bar] {
```

Comment:
Could you also add a test for a type parameter in contravariant position, e.g.
```scala
trait Service[-R] {
  def r: RIO[R, Unit]
}
```
?

Reply:
Added a test. Had to update the TypeDef modifiers for Service type params to avoid the generation of def foo[-R], etc.
I'm not sure if it is efficient enough.